### ISSUE TITLE
Sampling offset doesn't work at certain scales

### ISSUE THREAD

**Issue by tdbgamer** [StarArawn/bevy_ecs_tilemap#123]
I noticed while adjusting the camera's scale that there is an issue with the sampling when zoomed in at different levels. Seems to happen mostly when zoomed out:
![tilemap_sampling_issue](https://user-images.githubusercontent.com/3655689/145880726-d92e828b-06b2-4e6f-827e-8ae68eab1c3c.gif)

It seems like this line was an attempt to fix it:

https://github.com/StarArawn/bevy_ecs_tilemap/blob/d7c4f8c8aa7775061f38aaf22bf2040610012f31/src/render/square-tilemap.vert#L113

Back when I tried to implement this myself, this seemed to fix my problem (adding 1px of padding to the textureatlas):
https://github.com/bevyengine/bevy/issues/1949#issuecomment-825637328

So I expect the padding calculations done here will fix the bug:
https://github.com/bevyengine/bevy/blob/de8edd3165c379e05aabb38359b3f4b97f46540a/crates/bevy_sprite/src/texture_atlas.rs#L125-L159

**Comment by StarArawn** [comment]
We have existing padding support already. Checkout the tile spacing field here:
https://github.com/StarArawn/bevy_ecs_tilemap/blob/main/src/layer.rs#L47

A better solution to this problem is to not rely on texture atlases which are prone to bugs like this. In the far distant future, when I get some time, `bevy_ecs_tilemap` will no longer support atlases directly and instead internally everything will use a texture atlas instead. This will allow us to use texture filtering as well as mip maps which will great increase the quality of tilemap rendering when zooming.

**Comment by tdbgamer** [comment]
@StarArawn Yeah I tried adding padding in that field and it seems to completely mess up the sampling at that point.
This is what happens when I add `Vec2::new(1.0, 1.0)` of spacing.
![image](https://user-images.githubusercontent.com/3655689/145883126-81292741-835e-4a92-a5c0-a1cee3438802.png)

**Comment by StarArawn** [comment]
I would suggest we look at fixing this in the new renderer since it involves only fixing 1 spot. I can try to look at this tonight. 🤔

**Comment by tdbgamer** [comment]
Thanks! I'll try to look into it too, but I'm a noob with graphics programming and bevy internals for that matter. Very cool what you've done here though, way better than my naive approach when I did it manually.

**Comment by gerasim13** [comment]
![Screen Recording 2023-07-08 at 00 12 17](https://github.com/StarArawn/bevy_ecs_tilemap/assets/842316/6c15c0bc-81fa-4c40-9776-9a4918e20b42)
I have the same issue
